jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/providers/ScanningAppProvider.java

```diff
@@ -180,19 +181,19 @@ else if (coreProvider)
         if (StringUtil.isNotBlank(environmentName))
         {
             // If the app specifies the environment for this provider, then this deployer will deploy it.
-            if (Objects.equals(environmentName, getEnvironmentName()))
+            if (environmentName.equalsIgnoreCase(getEnvironmentName()))
             {
                 if (LOG.isDebugEnabled())
                     LOG.debug("{} created {}", this, app);
                 return app;
             }
 
             // If we are the default provider then we may warn
-            if (Objects.equals(getEnvironmentName(), _deploymentManager.getDefaultEnvironmentName()))
+            if (defaultEnvironmentName != null && defaultEnvironmentName.equalsIgnoreCase(getEnvironmentName()))
```

Comment:
Assuming for this provider the env name is not blank, then:

```suggestion
            if (getEnvironmentName().equalsIgnoreCase(defaultEnvironmentName))
```

Reply:
can't assume it is not null.

Reply:
Why not? What would this provider deploy then, if not one among core or eeN?

Reply:
Because I have to read several other methods to work out that at the time this method is called that the originally null field will probably not be null now. But whatever.